core/src/main/java/org/infinispan/jmx/CacheJmxRegistration.java

```diff
@@ -85,52 +95,59 @@ public void stop() {
       // If removing cache, also remove cache MBean
       if (unregisterCacheMBean) {
          globalJmxRegistration.unregisterCacheMBean(cacheName, cacheConfiguration.clustering().cacheModeString());
+         cacheMBeanRegistered = false;
       }
-
-      // make sure we don't set cache to null, in case it needs to be restarted via JMX.
    }
 
+   /**
+    * Indicates that the cache MBean is to be unregistered on stop. This is not normally done because we keep he MBean
+    * around to allow restart via jmx).
+    *
+    * @param unregisterCacheMBean
+    */
    public void setUnregisterCacheMBean(boolean unregisterCacheMBean) {
       this.unregisterCacheMBean = unregisterCacheMBean;
    }
 
    @Override
-   protected ComponentsJmxRegistration buildRegistrar() {
-      // Quote group name, to handle invalid ObjectName characters
-      String groupName = CACHE_JMX_GROUP + "," + globalJmxRegistration.getCacheJmxName(cacheName,
-            cacheConfiguration.clustering().cacheModeString()) + ",manager=" + ObjectName.quote(globalConfig.cacheManagerName());
-
-      if (!globalConfig.globalJmxStatistics().enabled() && jmxDomain == null) {
-         String tmpJmxDomain = JmxUtil.buildJmxDomain(globalConfig.globalJmxStatistics().domain(), mBeanServer, groupName);
-         synchronized (globalJmxRegistration) {
-            if (globalJmxRegistration.jmxDomain == null) {
-               if (!tmpJmxDomain.equals(globalConfig.globalJmxStatistics().domain()) && !globalConfig.globalJmxStatistics().allowDuplicateDomains()) {
-                  throw log.jmxMBeanAlreadyRegistered(tmpJmxDomain, globalConfig.globalJmxStatistics().domain());
-               }
-               // Set manager component's jmx domain so that other caches under same manager
-               // can see it, particularly important when jmx is only enabled at the cache level
-               globalJmxRegistration.jmxDomain = tmpJmxDomain;
-            }
-            // So that all caches share the same domain, regardless of whether dups are
-            // allowed or not, simply assign the manager's calculated jmxDomain
-            jmxDomain = globalJmxRegistration.jmxDomain;
+   public boolean isJMXEnabled() {
+      return cacheConfiguration.jmxStatistics().available();
+   }
+
+   @Override
+   protected String initGroup() {
+      return getCacheGroupName(cacheName, cacheConfiguration.clustering().cacheModeString(), globalConfig.cacheManagerName());
+   }
+
+   @Override
+   protected String initDomain() {
+      if (!globalJmxRegistration.isJMXEnabled()) {
+         if (globalJmxRegistration.mBeanServer == null) {
+            globalJmxRegistration.mBeanServer = mBeanServer;
```

Comment:
Why do we set the global jmx `mBeanServer` to the cache's mBeanServer if JMX is disabled globally?